sympy/calculus/util.py

```diff
@@ -61,15 +63,18 @@ def continuous_domain(f, symbol, domain):
     try:
         sings = S.EmptySet
         if f.has(Abs):
-            sings = solveset(1/f, symbol, domain)
+            sings = solveset(1/f, symbol, domain) + \
+                solveset(denom(together(f)), symbol, S.Reals)
         else:
             for atom in f.atoms(Pow):
-                predicate, denom = _has_rational_power(atom, symbol)
-                if predicate and denom == 2:
-                    sings = solveset(1/f, symbol, domain)
+                predicate, denomin = _has_rational_power(atom, symbol)
+                if predicate and denomin == 2:
+                    sings = solveset(1/f, symbol, domain) +\
+                        solveset(denom(together(f)), symbol, S.Reals)
                     break
             else:
-                sings = Intersection(solveset(1/f, symbol), domain)
```

Comment:
also in general, I think we could use `solveset(1/f, symbol, domain)` instead of using explicit intersection.

Reply:
The explicit intersection seems intentional. While solving for certain functions using `solveset`, for e.g. `1/tan(x)`, the function returns an ImageSet union if the domain is `S.Complexes` or `S.Reals`, but returns an empty set in bounded domains like `Interval(0, pi)`. Solving the function in the complex plain and taking the intersection with the domain solves this problem.